stellar-contract-env-host/src/host.rs

```diff
@@ -389,6 +397,42 @@ impl Host {
         // Resolve the function symbol and invoke contract call
         vm.invoke_function_raw(self, SymbolStr::from(func).as_ref(), args)
     }
+
+    pub fn invoke_function(&mut self, hf: HostFunction, args: ScVec) -> Result<ScVal, HostError> {
+        let mut args_iter = args.to_vec().into_iter();
+
+        let contract_id = match args_iter.next() {
+            None => Err(HostError::General("no contract id")),
+            Some(ScVal::Object(Some(ob))) => self.to_host_obj(&ob),
+            _ => Err(HostError::General("bad contract id")),
+        }?;
+
+        let symbol: Symbol = match args_iter.next() {
+            None => Err(HostError::General("no symbol")),
+            Some(ScVal::Symbol(s)) => s
+                .as_vec()
+                .as_slice()
+                .try_into()
+                .map_err(|e| HostError::General("bad symbol")),
+            _ => Err(HostError::General("bad symbol")),
+        }?;
+
+        let mut raw_args: Vec<RawVal> = Vec::new();
+        for scv in args_iter {
+            raw_args.push(self.to_host_val(&scv)?.val);
+        }
```

Comment:
Fyi if you wanted to avoid the mutable vec, you could write this using the map function. Although this is just a matter of taste, just mentioning it.
```suggestion
        let raw_args = args_iter
            .map(|scv| self.to_host_val(&scv).map(|ev| ev.val))
            .collect::<Result<Vec<RawVal>, HostError>>()?;
```